restaurants/src/app/components/restaurants/restaurants.component.ts

```diff
@@ -1,41 +1,43 @@
 import { Component, Input, OnInit } from "@angular/core";
 import { Restaurant } from "../../models/restaurants.model";
-import { RESTAURANTS } from "src/app/services/restaurants.service";
 import { throwIfEmpty } from "rxjs";
+import { v4 as uuidv4 } from 'uuid';
+import { RestaurantsService } from "src/app/services/restaurants.service";
 
 @Component({
     selector: 'app-restaurants',
     templateUrl: './restaurants.component.html',
     styleUrls: ['./restaurants.component.scss']
   })
-  export class RestaurantsComponent {
-    Restaurant = RESTAURANTS;
+  export class RestaurantsComponent implements OnInit {
+    constructor(private restaurantsService: RestaurantsService){}
+    restaurants: {[id: string]: Restaurant} = {};
+
     name: string = '' ;
     rank: number = 0;
     id: number = 0;
 
-    newRestaurant: Restaurant = {id: 0, name: '', rank: 0};
+    newRestaurant: Restaurant = { name: '', rank: 0};
 
-    selectedRestaurant?: Restaurant;
-    onSelect(restaurant: Restaurant): void{
-      this.selectedRestaurant= restaurant;
+    selectedRestaurant?: string;
+    onSelect(restaurantKey: string): void{
+      this.selectedRestaurant = restaurantKey;
+    }
+    ngOnInit() {
+      this.restaurants = this.restaurantsService.getAll();
     }
 
-    add(newId: number, newName: string, newRank: number){
-
-      this.newRestaurant = {id: newId, name: newName, rank: newRank};
-      RESTAURANTS.push(this.newRestaurant);
+    add(name: string, rank: number){
+      this.restaurantsService.add(name, rank);
```

Comment:
Does this method need to take in the parameters of `name` and `rank`? I see in your html that the inputs are bound via `ngModel` to the `name` and `rank` variables - so could those parameters be removed and line 31 just be `this.restaurantsService.add(this.name, this.rank);` instead?

Reply:
yes